sql/timetype.go

```diff
@@ -0,0 +1,284 @@
+package sql
+
+import (
+	"fmt"
+	"gopkg.in/src-d/go-errors.v1"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/proto/query"
+)
+
+var (
```

Comment:
It's a little strange / inconsistent that this top-level type file doesn't have a new interface type in it. Why is that? Seems like there should be a Time interface, even if just embeds the other datetime interface.